src/components/molecules/PostLinkEmbed/PostLinkEmbed.tsx

```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import * as Atoms from '@/atoms';
+import { useMemo } from 'react';
+
+export type PostLinkEmbedProps = {
+  content: string;
+};
+
+type LinkEmbed = 'none' | 'youtube';
+
+const parseContentForLinkEmbed = (content: string): { type: LinkEmbed; url: string } => {
```

Comment:
At the top of the file or in a types file. Then the function signature becomes cleaner.

```suggestion
type EmbedType = 'youtube' | 'vimeo' | 'twitch' | 'twitter' | 'generic';
type LinkEmbed = { type: EmbedType | 'none'; url: string };

const parseContentForLinkEmbed = (content: string): LinkEmbed => {
```